src/lib/environment/crateinfo.rs

```diff
@@ -194,8 +195,18 @@ struct CargoConfig {
 #[derive(Debug, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 struct CargoConfigBuild {
+    #[serde(default, deserialize_with = "deserialize_target")]
     target: Option<String>,
-    target_dir: Option<String>,
+    target_dir: Option<PathBuf>,
+}
+
+fn deserialize_target<'de, D: Deserializer<'de>>(
+    deserializer: D,
+) -> Result<Option<String>, D::Error> {
+    Option::<String>::deserialize(deserializer)?
+        .map(|target| target.trim_end_matches(".json").to_string())
```

Comment:
why do we need it in the deserialize? why not keep it simple and do it in the usage as today?

Reply:
'Cause it will lead us to code repetition. We need to trim string in `crate_target_dir` and `crate_target_triple`, but original string can be useful for env var like `CARGO_MAKE_IS_CUSTOM_TARGET_TRIPLE`